catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java

```diff
@@ -88,7 +116,29 @@ protected SolrCacheMBean getCacheProxy() {
 
         String[] idsArray = new String[ids.size()];
         idsArray = ids.toArray(idsArray);
-        verify(mbean, times(1)).removeById(idsArray);
+        verify(solrCacheMock, times(1)).removeById(idsArray);
+    }
+
+    /**
+     * Tests condition where number of results returned by each catalog framework query
+     * is more than batchSize being used.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testQueryGreaterThanBatchSize() throws Exception {
+
+        totalResultsToDelete = 255;
+        String expectedResult = totalResultsToDelete + " metacards successfully deleted.";
+        filter = "title like '*'";
+
+        configureMocks(250, 5, 0);
+        setUpRemoveCommand();
+        removeCommand.executeWithSubject();
+
+        String actualResult = consoleOutput.getOutput();
+        assertThat(actualResult, containsString(expectedResult));
+        verify(catalogFrameworkMock, times(2)).delete(isA(DeleteRequest.class));
```

Comment:
❗️For the new tests, we may want to capture the parameter being passed to `delete` using an `ArgumentCaptor` and at least make sure we're calling the catalog framework with the right number of IDs.